components/molecules/HighlightInput/highlight-input-form.tsx

```diff
@@ -258,14 +283,30 @@ const HighlightInputForm = ({ refreshCallback }: HighlightInputFormProps): JSX.E
       }
     };
 
+    const removeAlreadyHighlightedSuggestions = async (newHighlightSuggestions: any[]) => {
+      // get all the highlights of the user
+      const allHighlights = await fetchAllUserHighlights(1);
+      console.log("allHighlights", allHighlights);
+
+      // get all the urls of the highlights
+      const allHighlightUrls = allHighlights.map((highlight) => highlight.url);
+
+      // remove any suggestions that have already been highlighted
+      const filteredSuggestions = newHighlightSuggestions.map((suggestionPage) =>
+        suggestionPage.filter((suggestion: { url: string }) => !allHighlightUrls.includes(suggestion.url))
+      );
+
+      setHighlightSuggestions(filteredSuggestions);
+    };
```

Comment:
I thought It would be nice to have this function here return a filtered array of suggestions that hasn't been highlighted yet.

This IMO will help reduce the friction of setting the state of `highlightSuggestions` twice... as seen in the next comment..

Reply:
Thanks for pointing this out, missed it while iterating.